vortex-array/src/array/chunked/canonical.rs

```diff
@@ -28,6 +28,8 @@ pub(crate) fn try_canonicalize_chunks(
     chunks: Vec<Array>,
     dtype: &DType,
 ) -> VortexResult<Canonical> {
+    assert!(!chunks.is_empty(), "chunks must be non-empty");
```

Comment:
is it crash-worthy programmer error if it's empty? if not, should return an `Err` instead

Reply:
done